### ISSUE TITLE
Question: Breaking changes in 1.5.6.2

### ISSUE THREAD

**Issue by mscrivo** [SpringMT/zstd-ruby#78]
Hey @SpringMT, could you clarify what these breaking changes mean from a user's perspective?

```
Zstd.compress uses ZSTD_compressStream2 instead of ZSTD_compress
Zstd.decompress uses ZSTD_decompressDCtx instead of ZSTD_decompress
```

**Comment by SpringMT** [comment]
Apologies for any confusion regarding the breaking changes.

Firstly, the `Zstd.compress` and `Zstd.decompress` methods in version v1.5.6.2 are compatible with other versions, meaning they can be used in the same manner as before. 
Thus, these changes are more along the lines of Improvements rather than Breaking Changes.

Here's a detailed explanation:

Regarding the recent modifications, there are two main aspects:

1. Unified dictionary specification for both Simple and Streaming Compression/Decompression:

For compression, I have standardized the handling of dictionaries by using `ZSTD_compressStream2`.

`ZSTD_compress` internally calls `ZSTD_compressCCtx`, which in turn calls `ZSTD_compress_usingDict`.
This method is compatible with a new method `ZSTD_compress2`.
See: [GitHub Issue #3711](https://github.com/facebook/zstd/issues/3711)
`ZSTD_compress2` internally uses `ZSTD_compressStream2`, making it compatible with `ZSTD_compress`. `ZSTD_compressStream2` allows dictionaries to be specified using `ZSTD_CCtx_setParameter` method, thus enabling the switching of methods used for compression without changing the context handling part.
This makes loading dictionaries easier and enhances maintainability by unifying the methods used for ZSTD compression and handling dictionaries.

For decompression, I have standardized the handling of dictionaries by using `ZSTD_decompressDCtx`.
Previously used `ZSTD_decompress` internally used `ZSTD_decompressDCtx`.
I considered using `ZSTD_decompressStream` (used by Streaming Decompression currently) both Simple Decompression and Streaming Decompression but decided against it due to performance concerns.

2. Preparation for multi-threading support and removing GVL:

I am working on reducing the code change for unlocking GVL during compression and decompression processes. This change is currently being implemented.
